### PR TITLE
electron - limit crash reporter to few insiders users

### LINKED ISSUE #120902: Electron: crash reporter slows down startup
There is a new perf mark to measure the impact of the crash reporter on the startup which is a sync blocking call. 

Even on my fast macOS ARM machine, this seems to consume a considerable amount of time:

image

Given the crash reporter does not provide any async API, I see no good way of fixing this unless we can start the crash reporter after the window has opened, which I think is not an option?

//cc @jrieken

### PR DESCRIPTION
**Pull request by bpasero**
This PR fixes #120902 by limiting the crash reporter to:
* insiders only
* users who's crash reporter ID ends with `0`

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -336,19 +337,24 @@ function configureCrashReporter() {
...
-			if (uuidPattern.test(crashReporterId)) {
-				if (isWindows) {
+		const crashReporterId = argvConfig['crash-reporter-id'];
+		const uuidPattern = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
+		if (
+			appCenter &&							// only when appCenter is configured
+			product.quality === 'insider' &&		// only for insiders
+			typeof crashReporterId === 'string' &&	// only if we have a crash reporter ID
+			uuidPattern.test(crashReporterId) &&	// only if the crash reporter ID is valid
+			crashReporterId.endsWith('0')			// only if the crash reporter ID ends with `0`
```
Comment (bpasero) [inline src/main.js:354]: This is a heuristic I picked, but maybe you have a better idea.

File: src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts [hunk 2]
```diff
@@ -214,28 +214,34 @@ export class LocalProcessExtensionHost implements IExtensionHost {
...
-					const crashReporterId = this._environmentService.crashReporterId; // crashReporterId is set by the main process only when crash reporting is enabled by the user.
-					const appcenter = this._productService.appCenter;
-					const uploadCrashesToServer = !this._environmentService.crashReporterDirectory; // only upload unless --crash-reporter-directory is provided
-					if (uploadCrashesToServer && appcenter && crashReporterId && isUUID(crashReporterId)) {
-						const submitURL = appcenter[`linux-x64`];
-						crashReporterStartOptions.submitURL = submitURL.concat('&uid=', crashReporterId, '&iid=', crashReporterId, '&sid=', crashReporterId);
-						crashReporterStartOptions.uploadToServer = true;
+					const crashReporterDirectory = this._environmentService.crashReporterDirectory;
+					const crashReporterId = this._environmentService.crashReporterId;
+					if (crashReporterDirectory || crashReporterId) {
```
Comment (bpasero) [inline src/vs/workbench/services/extensions/electron-browser/localProcessExtensionHost.ts:223]: Now using these properties as a hint whether crash reporting is enabled at all.

**Comment by sadeghzare2580** [comment]
> This PR fixes #120902 by limiting the crash reporter to:
> * insiders only
> * users who`s crash reporter ID ends with `0`
>

File: src/main.js (hunk 1 above)
Comment (bpasero) [reply src/main.js:354]: @deepak1556 I ran some experiments generating `30000` UUIDs to roughly simulate our number of present insider users and typically end up with a number around `~1900` that match. Would you think that number if sufficient to find crashes?

**Review by deepak1556 — CHANGES_REQUESTED** [review]
Discussions pending offline related to some logistic concerns with disabling the reporter in stable.

**Pull request closed without merging**